### ISSUE TITLE
Submit only changed values or values that are not pristine

### ISSUE THREAD

**Issue by eventhough** [christianalfoni/formsy-react#133]
When we submit a form there should be an option for us to get only the inputs that have changed or are not pristine. This is useful for a database update where we only need to update columns that have changed.

**Comment by ppiekarczyk** [comment]
:+1:

**Comment by aphillipo** [comment]
This sounds amazing.  Means you are less likely to have multiple people editing the same thing clobber eachothers results.

**Comment by christianalfoni** [comment]
Hi guys,

Yes, this is indeed something nice to have, though I am unsure that this job is for Formsy. I would expect this scenario has a model with the state of some entity on the server, like a person. The model is presented in a formsy form. Deciding what to pass to the server I think is the responsibility of the model layer, not the UI layer. So you would rather pass the whole form to the model-layer, figure out what is actually necessary to pass to the server and then do that. If Formsy is in control of this you will most likely meet scenarios where Formsy has some logic deciding what to pass to the server, and the model layer might have some of the other logic, maybe Formsy does not do a good enough job and we have to build upon the API to solve lots of different scenarios.

So yeah, I am concerned about introducing it in Formsy, as this is closer to the model layer than the UI layer I think. It basically drills down to comparing two objects and extracting the diff, which some other lib probably already does  very effectively.

But please, if you disagree we can discuss it more :-) Just my initial thoughts!

**Comment by aphillipo** [comment]
Yeah I suppose it works both ways doesn't it (do it on the backend or the front end is a marginal tradeoff), probably you could say it's one for the backend, but then over the wire transmission should aim to be as small as possible.

I say keep formsy as simple as is possible, but then it should not be so difficult to have an object of bits to say which fields have changed and _.filter the stuff passed to submit.

**Comment by christianalfoni** [comment]
Yeah, though it depends on how you implement it. Either Formsy needs to keep track on "changed values" or it does a "diffing" of initial form values and current values when submitting.

The last part if the simplest and most effective implementation I would suggest... that said, it is very easy to make yourself. Often libraries like Backbone Model or Angular Resource allows you to do this very easily.

Sorry, but I think it is the correct choice to have this logic closer to where you communicate with the server, not inside Formsy. But please, if a lot of people disagree I will of course reconsider.

**Comment by aphillipo** [comment]
Well the argument is that formsy probably:

a) knows what the original data was
b) knows which fields are no longer "pristine"
c) the difference being sent over the wire is "cheaper" than everything being sent

I think this should be possible to build separately from formsy though - there should be an easy way to get this info and do the diff??  

I'm massively worried by this in general though.  Bigger fish to fry ;-)

**Comment by christianalfoni** [comment]
Let me keep this open and we will revisit after next release

**Comment by mbrookes** [comment]
@aphillipo you've mentioned back-end v front-end, and network bandwidth concerns, but front-end doesn't mean this has to be performed by formsy, your onValidSubmit method can take care of it client side, so your "over-the-wire" is minimised, without it being a function of formsy. 

Having said that, I do agree that there's an argument to be made for an option for formsy to only pass changed properties - there are enough use cases for that to be useful as a binary option, without overloading the API.
